sheetstorm/tests/test_unit_generate_test_data.py

```diff
@@ -0,0 +1,259 @@
+
+    def test_get_superuser_request_function_should_return_false_if_superuser_already_exists(self):
+        UserFactory(is_superuser=True)
+
+        self.assertFalse(GenerateTestDataCommand._get_superuser_request({self.superuser: True}))
+
+    @parameterized.expand([CustomUser.UserType.ADMIN.name, superuser])
+    def test_admin_and_superuser_should_have_staff_property(self, user_type):
+        self.assertEqual(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("is_staff"), True)
+
+    @parameterized.expand([CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name])
+    def test_employee_and_manager_should_not_have_staff_property(self, user_type):
+        self.assertEqual(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("is_staff"), False)
+
+    @parameterized.expand(
+        [CustomUser.UserType.ADMIN.name, CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name]
+    )
+    def test_any_user_should_not_have_superuser_property_except_superuser(self, user_type):
+        self.assertEqual(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("is_superuser"), False)
+
+    def test_superuser_should_have_superuser_property(self):
+        self.assertEqual(GenerateTestDataCommand._set_user_factory_parameters(self.superuser).get("is_superuser"), True)
+
+    @parameterized.expand([CustomUser.UserType.ADMIN.name, superuser])
+    def test_admin_and_superuser_should_have_user_type_coded_as_admin(self, user_type):
+        self.assertEqual(
+            GenerateTestDataCommand._set_user_factory_parameters(user_type).get("user_type"),
+            CustomUser.UserType.ADMIN.name,
+        )
+
+    @parameterized.expand([CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name])
+    def test_employee_and_manager_should_have_user_type_coded_as_based_of_user_type_name(self, user_type):
+        self.assertEqual(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("user_type"), user_type)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 2),
+            (CustomUser.UserType.EMPLOYEE.name, 5),
+            (CustomUser.UserType.MANAGER.name, 3),
+        ]
+    )
+    def test_result_of_create_user_function_should_be_specified_number_of_users_in_database(
+        self, user_type, number_of_users_to_create
+    ):
+        GenerateTestDataCommand().create_user(user_type, number_of_users_to_create)
+
+        self.assertEqual(CustomUser.objects.filter(user_type=user_type).count(), number_of_users_to_create)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 2, 2),
+            (CustomUser.UserType.EMPLOYEE.name, 4, 3),
+            (CustomUser.UserType.MANAGER.name, 2, 1),
+        ]
+    )
+    def test_create_user_function_should_not_create_users_if_number_in_database_is_greater_than_or_equal_to_specified(
```

Comment:
Why not including similar test for projects? 🙂

Reply:
Done